src/views/Invite/GenerateInvites.tsx

```diff
@@ -0,0 +1,167 @@
+  useInviteStore,
+} from './useInvites';
+import { DEFAULT_NUM_INVITES } from 'lib/constants';
+import { useTimerStore } from 'store/timerStore';
+import { useWallet } from 'store/wallet';
+
+export const GenerateInvites = () => {
+  const { pop } = useLocalRouter();
+  const { walletType }: any = useWallet();
+  const { point, nextQuotaTime } = useRollerStore();
+  const { getPendingTransactions, getAndUpdatePoint } = useRoller();
+  const { nextRoll } = useTimerStore();
+  const { inviteJobs } = useInviteStore();
+  const { generateInviteCodes } = useInvites();
+  const [generatingStatus, setGeneratingStatus] = useState<
+    InviteGeneratingStatus
+  >('initial');
+  const [error, setError] = useState('');
+  const currentL2 = Boolean(point.isL2Spawn);
+  const [numInvites, setNumInvites] = useState(
+    currentL2 ? DEFAULT_NUM_INVITES : 1
+  );
+
+  const createInvites = useCallback(async () => {
+    setGeneratingStatus('generating');
+    try {
+      await generateInviteCodes(numInvites);
+      getPendingTransactions();
+      getAndUpdatePoint(point.value);
+      setGeneratingStatus('finished');
+    } catch (error) {
+      if (typeof error === 'object' && (error as Error)?.message) {
+        setError((error as Error).message);
+      } else if (typeof error === 'string') {
+        setError(error);
+      } else {
+        setError('Generating invites failed');
+      }
+      setGeneratingStatus('errored');
+    }
+  }, [
+    generateInviteCodes,
+    numInvites,
+    getPendingTransactions,
+    getAndUpdatePoint,
+    point,
+    setGeneratingStatus,
+  ]);
+
+  const overQuota = numInvites > point.l2Quota && numInvites > 0;
+  const generateDisabled = overQuota; //loading?
+  const showTxnNote = isExternalWallet(walletType);
+
+  return (
+    <View
+      pop={pop}
+      className="cohort show-invite-form"
+      hideBack
+      header={<L2BackHeader back={pop} />}>
+      <Window>
+        <HeaderPane>
+          <h5>Generate Invite Codes</h5>
+        </HeaderPane>
+        <BodyPane>
+          <Box className="upper">
+            {currentL2 ? (
+              <Row className="points-input">
+                I want to generate
+                <StatelessTextInput
+                  className={`input-box ${overQuota ? 'over-quota' : ''}`}
+                  value={numInvites}
+                  maxLength={3}
+                  onChange={e => {
+                    const target = e.target as HTMLInputElement;
+                    setNumInvites(Number(target.value.replace(/\D/g, '')));
+                  }}
+                />
+                planet invite code{numInvites > 1 ? 's' : ''}
+              </Row>
+            ) : (
+              <Box className="migration-prompt">
+                ETH fees are very high currently. You can spawn invites for free
+                after{' '}
+                <span className="migrate" onClick={() => null}>
+                  {/* TODO: change this to navigate to migration */}
```

Comment:
Is this still needed?

Reply:
I forgot to rip it out after our discussion in the bridge channel. I'll do that and re-request a review.